spec/controllers/items_controller_spec.rb

```diff
@@ -2,21 +2,40 @@
 
 RSpec.describe ItemsController, type: :controller do
   describe "GET new" do
-    it "get form for new item" do
+    it "redirects when not logged in" do
+      get :new
+      expect(response).to redirect_to(login_path)
+    end
+
+    it "get form for new item when logged in" do
+      session[:admin_id] = "123"
+      allow(Admin).to receive(:find_by).with({ :id => "123" }).and_return(true)
+
       get :new
       expect(response).to render_template(:new)
       expect(assigns(:item)).to be_present
     end
   end
 
   describe "POST create" do
+    it "redirects when not logged in" do
+      post :create, item:{ name: "item1", price: 1.2 }
+      expect(response).to redirect_to(login_path)
+    end
+
     it "creates new item when valid" do
+      session[:admin_id] = "123"
+      allow(Admin).to receive(:find_by).with({ :id => "123" }).and_return(true)
```

Comment:
We tend to prefer the newer hash syntax `id: 123` over the older hashrocket syntax. You'll still see it around in older codebases, but we've mostly cleaned it up using rubocop in current projects. Also, in this case, you don't need the curly braces.